src/Build.UnitTests/BackEnd/FailingBuilderTask.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Build.UnitTests
+{
+    public  class FailingBuilderTask : Task
+    {
+        public FailingBuilderTask()
+            : base(null)
+        { }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override bool Execute()
+        {
+            try
+            {
+                BuildEngine.BuildProjectFile(CurrentProject, new string[] { "CoreCompile" }, new Hashtable(), null);
```

Comment:
Comment that this calling pattern is like the XAML thing we care about specifically.